functional/push-attestation-on-localhost/test.sh

```diff
@@ -0,0 +1,125 @@
+
+        # Set the configuration for the agent
+        rlRun "limeUpdateConf agent measuredboot_ml_path '\"/var/tmp/binary_bios_measurements\"'"
+        # TODO: this is not used anywhere
+        #rlRun "limeUpdateConf agent uefi_logs_binary_path '\"/var/tmp/binary_bios_measurements\"'"
+
+        # Copy the fake UEFI log
+        rlRun "cp binary_bios_measurements /var/tmp"
+
+        # Disable EK certificate verification on the tenant
+        rlRun "limeUpdateConf tenant require_ek_cert False"
+
+        # Configure TPM emulator if needed
+        if limeTPMEmulated; then
+            rlRun "limeStartTPMEmulator"
+            rlRun "limeWaitForTPMEmulator"
+            rlRun "limeCondStartAbrmd"
+            rlRun "limeInstallIMAConfig"
+            rlRun "limeStartIMAEmulator"
+        fi
+
+        sleep 5
+
+        # Start keylime services with push support
+        rlRun "limeStartVerifier"
+        rlRun "limeWaitForVerifier"
+        rlRun "limeStartRegistrar"
+        rlRun "limeWaitForRegistrar"
+        # Start push-attestaton agent
+        rlRun "limeStartPushAgent"
+        rlRun "limeWaitForAgentRegistration ${AGENT_ID}"
+
+        # create some scripts
+        TESTDIR=$(limeCreateTestDir)
+        rlRun "echo -e '#!/bin/bash\necho This is good-script1' > $TESTDIR/good-script1.sh && chmod a+x $TESTDIR/good-script1.sh"
+        rlRun "echo -e '#!/bin/bash\necho This is good-script2' > $TESTDIR/good-script2.sh && chmod a+x $TESTDIR/good-script2.sh"
+
+        # create allowlist and excludelist
+        rlRun "limeCreateTestPolicy ${TESTDIR}/*"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Add keylime agent"
+        REVOCATION_SCRIPT_TYPE=$( limeGetRevocationScriptType )
+        rlRun "cat > script.expect <<_EOF
+set timeout 20
+spawn keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u $AGENT_ID --verify --runtime-policy policy.json --include payload-${REVOCATION_SCRIPT_TYPE} --cert default -c add --push-model
+expect \"Please enter the password to decrypt your keystore:\"
+send \"keylime\n\"
+expect eof
+_EOF"
+        rlRun "expect script.expect"
+        # Check that agent appears in verifier's agent list
+        rlRun -s "keylime_tenant -c cvlist"
+        # shellcheck disable=SC2154  # rlRun_LOG is set by BeakerLib's rlRun -s
+        rlAssertGrep "$AGENT_ID" "$rlRun_LOG"
+
+        # TODO: For now the agent dies after starting because it tries to send
+        # measurements and fails. Restart the agent.
+        rlRun "limeStartPushAgent"
+        rlRun "limeWaitForAgentRegistration ${AGENT_ID}"
+
+        rlAssertGrep "PATCH Response code:.*202 Accepted" "$(limePushAgentLogfile)" -E
+        rlAssertGrep "Attestation [0-9]+ for agent '${AGENT_ID}' verified successfully" "$(limeVerifierLogfile)" -E
+
+        # Store the index of the first attestation
+        INDEX=$(grep -oE "Attestation [0-9]+ for agent '${AGENT_ID}' verified successfully" "$(limeVerifierLogfile)" |
```

Comment:
It is not visible because no match equals 0 but the correct form is
```
INDEX=$(grep -oE "Attestation [0-9]+ for agent .${AGENT_ID}. verified successfully" "$(limeVerifierLogfile)" | tail -1 |  grep -oE "Attestation [0-9]+" | grep "[0-9]+")
```

Reply:
Nice catch, thanks!